### PR TITLE
Add minimal example using Casbin to Get and Create Policies

### PR DESCRIPTION
**Pull request by LucasHartridge**

### CODE REVIEW

File: authorization/casbin.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from casbin.enforcer import Enforcer
+from casbin.persist.adapters import FileAdapter
+
+
+class Casbin():
```
Comment (jcunhafonte) [inline authorization/casbin.py]: You don’t need to construct if you are not extending it.

File: authorization/casbin.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    """
+    Casbin class that receives the location of the casbinmodel or it loads the default one
+    """
+
+    def __init__(self, model: str = "authorization/casbinmodel.conf"):
+        self.enforcer = Enforcer(model, FileAdapter('authorization/ac_policy.csv'))
+
+    
+    def get_users():
```
Comment (jcunhafonte) [inline authorization/casbin.py]: Create a user class.

File: permission.py [hunk 3]
```diff
@@ -1,50 +1,33 @@
...
 
 
 class PermissionServicer(permission_pb2_grpc.PermissionServicer):
     def GetPolicies(self, request, context):
-        print("GetPolicies Request Made:")
-        print(request)
+        if request.user_id not in Casbin.get_users():
+            raise grpc.RpcError(grpc.StatusCode.NOT_FOUND, 'User not found')
 
-        user = users.get(request.user_id)
```
Comment (jcunhafonte) [inline permission.py:40]: This probably should be a decorator.

File: reporting.py [hunk 4]
```diff
@@ -1,27 +1,36 @@
...
-            user_id = int(user_id)
-            policy_request = permission_pb2.PolicyRequest(user_id=user_id)
-
-            try:
-                policies_reply = stub.GetPolicies(policy_request)
-                print("GetPolicies Response Received:")
-                print(policies_reply)
-            except grpc.RpcError as e:
-                print(e)
+            actions = input_available_actions()
```
Comment (jcunhafonte) [inline reporting.py:14]: Use enum

**Review by jcunhafonte — CHANGES_REQUESTED** [review]

**Comment by LucasHartridge** [comment]
@jcunhafonte Ready for review!

File: authorization/user.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+from typing import List
+from functools import wraps
+from casbin.enforcer import Enforcer
+from casbin.persist.adapters import FileAdapter
+
+
+class User:
+    def __init__(self, id: int):
+        self.id = str(id)
+        self.enforcer = Enforcer("authorization/casbinmodel.conf", FileAdapter('authorization/ac_policy.csv'))
```
Comment (jcunhafonte) [inline authorization/user.py]: You are such a boss. I’d only avoid having enforcer here.

File: authorization/user.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    def add_policy(self, object: str, action: str):
+        self.enforcer.add_named_policy("p", self.id, object, action)
+        self.enforcer.save_policy()
+
+    def has_policy(self, object: str, action:str):
+        if self.enforcer.enforce(self.id, object, action):
+            return dict(status_code=200, message=f"User {self.id} has access!")
+
+        return dict(status_code=401, message=f"User {self.id} is not authorized")
```
Comment (jcunhafonte) [inline authorization/user.py]: Should be forbidden. But I also think we should return true or false. We are not handling http codes here.

**Review by jcunhafonte — CHANGES_REQUESTED** [review]

**Pull request merged**